vespajlib/src/main/java/com/yahoo/tensor/functions/Reduce.java

```diff
@@ -128,31 +128,42 @@ else if (argument.type().dimensions().size() == 1 && argument instanceof Indexed
         TensorType reducedType = outputType(argument.type(), dimensions);
 
         // Reduce cells
-        Map<TensorAddress, ValueAggregator> aggregatingCells = new HashMap<>();
+        int[] indexesToKeep = createIndexesToKeep(argument.type(), dimensions);
+        // TODO cells.size() is most likely an overestimate, and might need a better heuristic
+        // But the upside is larger than the downside.
+        Map<TensorAddress, ValueAggregator> aggregatingCells = new HashMap<>((int)argument.size());
```

Comment:
Consider checking for truncation here, if we support (ever) support large enough tensors for that to happen.

Reply:
See comment above